SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c

```diff
@@ -4165,13 +4165,28 @@ LoadSignatureData (
 STATIC EFI_STATUS
 EFIAPI
 KeyEnrollReset (
-  VOID
+  IN SECUREBOOT_CONFIGURATION        *ConfigData
   )
 {
   EFI_STATUS  Status;
   UINT8       SetupMode;
+  BOOLEAN     SecureBootEnable;
+  UINTN       DataSize;
 
-  Status = EFI_SUCCESS;
+  DataSize = sizeof (SecureBootEnable);
+  Status = gRT->GetVariable(
+                EFI_SECURE_BOOT_ENABLE_NAME,
+                &gEfiSecureBootEnableDisableGuid,
+                NULL,
+                &DataSize,
+                &SecureBootEnable
+                );
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "Cannot read SecureBootEnable variable: %r\n", Status));
+    /* Get the state from the from data if we failed to read the variable */
+    SecureBootEnable = ConfigData->AttemptSecureBoot;

```

Comment:
> Why not read from PCD if there is no variable? That way ConfigData wouldn't have to be passed to this function.

This is not about restoring the default, but preserving current state. We want the reset keys to not change SB state.

Reply:
> On a tangent note, as I can't add comments to unmodified code, I think the [test done in lines 3227-3229](https://github.com/Dasharo/edk2/blob/21462205e5e95e51f8f6b00da3c43e85da9c9cdf/SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c#L3227-L3229) should be moved outside of if ((*SetupMode) == USER_MODE). Right now the Enable Secure Boot checkbox isn't checked when SecureBootEnable is true, but there is no PK provisioned. There already is a [proper message about PK](https://github.com/Dasharo/edk2/blob/21462205e5e95e51f8f6b00da3c43e85da9c9cdf/SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigStrings.uni#L21) printed in that case, I think having this option grayed out but checked would be less confusing than having it empty.

Checked and grayed out even when PK is not provisioned? Is such a state even valid? Being unable to disable Secure Boot like that is a bad UX IMO.

> BTW is it normal that grayoutif is bold instead of gray on serial? It doesn't intuitively suggest that the option is disabled...

:shrug: